pages/api/graphql.js

```diff
@@ -56,8 +56,11 @@ export default startServerAndCreateNextHandler(apolloServer, {
     const apiKey = req.headers['x-api-key']
     let session
     if (apiKey) {
-      const sessionFieldSelect = { name: true, id: true, email: true }
-      const user = await models.user.findUnique({ where: { apiKey }, select: { ...sessionFieldSelect, apiKeyEnabled: true } })
+      const [user] = await models.$queryRaw`
+      SELECT id, name, email, "apiKeyEnabled"
+      FROM users
+      WHERE "apiKeyHash" = encode(digest(${apiKey}, 'sha256'), 'hex')
```

Comment:
I suppose this is not susceptible to SQLI due to using `models.$queryRaw`, vs `models.$queryRawUnsafe`, right?

Reply:
Yes, `$queryRaw` is safe to use with tagged templates.

It's not safe if called as a function with parentheses since you could have already injected SQL while building your string but that's not what I do here.[^1]

[^1]: https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries#in-queryraw-and-executeraw